### PR TITLE
Allow copy single file as indicated by tooltip

### PR DESCRIPTION
**Pull request by yacaovsnc**
When selecting "Artifact publish location: A file share", we do not support publish a file, even though the tooltip mentions "The folder or **file** path to publish".

Tested:
1. Path to publish set to a folder -- all files under this folder are copied
2. Path to publish set to a file with and without spaces -- the single file is copied in both cases.

### CODE REVIEW

File: Tasks/PublishBuildArtifacts/publishbuildartifacts.ts [hunk 1]
```diff
@@ -96,6 +97,11 @@ async function run() {
                 // copy the files
                 let script: string = path.join(__dirname, 'Invoke-Robocopy.ps1');
                 let command: string = `& ${pathToScriptPSString(script)} -Source ${pathToRobocopyPSString(pathtoPublish)} -Target ${pathToRobocopyPSString(artifactPath)} -ParallelCount ${parallelCount}`
+                if (fs.statSync(pathtoPublish).isFile()) {
+                    let parentFolder = path.dirname(pathtoPublish);
```
Comment (lkillgore) [inline Tasks/PublishBuildArtifacts/publishbuildartifacts.ts]: I think these `let`s can be `const`.

File: Tasks/PublishBuildArtifacts/Invoke-Robocopy.ps1 [hunk 2]
```diff
@@ -6,8 +6,11 @@ param(
     [Parameter(Mandatory = $true)]
     [string]$Target,
 
-	[Parameter(Mandatory = $true)]
-    [int]$ParallelCount)
+    [Parameter(Mandatory = $true)]
+    [int]$ParallelCount,
+
+    [Parameter(Mandatory = $false)]
```
Comment (madhurig) [inline Tasks/PublishBuildArtifacts/Invoke-Robocopy.ps1]: Why does this need to be an input?

Comment (yacaovsnc) [reply Tasks/PublishBuildArtifacts/Invoke-Robocopy.ps1]: This script is invoked by the typescript file, it's not run by the powershell handler directly.  We need to inform the PS1 script to know which file to copy.

File: Tasks/PublishBuildArtifacts/publishbuildartifacts.ts (hunk 1 above)
Comment (yacaovsnc) [reply Tasks/PublishBuildArtifacts/publishbuildartifacts.ts]: Cool.  Next time if I add new strings I will keep this in mind.  For now I think using "let" looks more consistent with the rest of the file.

File: Tasks/PublishBuildArtifacts/Invoke-Robocopy.ps1 (hunk 2 above)
Comment (madhurig) [reply Tasks/PublishBuildArtifacts/Invoke-Robocopy.ps1]: Ah, I see it now, strange!

**Pull request merged**